### PR TITLE
fix(pandas): Fix anti-joins

### PR DESCRIPTION
**Pull request by cwisecarver**
This PR fixes support for anti-joins in pandas. Currently a pandas anti-join will filter out a row if *any* of the columns match versus all of the columns matching. This changes the behavior to only remove rows where all columns match. This matches the behavior of the duckdb backend.

Happy to work with maintainers to get this merged. I tried to install Nix on my M1 mac to run the unit tests but lychee wouldn't compile on macOS even under rosetta2.

Potentially fixes: https://github.com/ibis-project/ibis/issues/4268

### CODE REVIEW

File: ibis/backends/tests/test_join.py [hunk 1]
```diff
@@ -17,7 +17,9 @@ def _pandas_semi_join(left, right, on, **_):
 def _pandas_anti_join(left, right, on, **_):
     assert len(on) == 1, str(on)
     inner = pd.merge(left, right, how="outer", indicator=True, on=on)
-    return inner["_merge"].apply(lambda x: True if x == "both" else False).astype(bool)
+    return inner["_merge"].apply(
+        lambda x: True if x == "both" else False
```
Comment (cpcloud) [inline ibis/backends/tests/test_join.py:21]: Would `inner["_merge"] == "both"` work here?

Comment (cwisecarver) [reply ibis/backends/tests/test_join.py]: That worked in my local test case that is included in the linked issue but failed CI with a different set of errors. Seems like progress at least.

Comment (cwisecarver) [reply ibis/backends/tests/test_join.py]: I think that change causes the length of the list to change (which caused the different errors) whereas mine just replaced everything with something. I'll try and find a more pandas-way to get the same result. I'm waiting for nix to finish installing on a linux box so I can hopefully run the tests locally.

Comment (cpcloud) [reply ibis/backends/tests/test_join.py]: FWIW you can use nix, conda or pip to develop with ibis.

**Comment by cpcloud** [comment]
The CI errors look unrelated to whether you use `apply`, and are related to a new warning from pandas about `sort_values` becoming keyword-only in the future.

**Comment by cwisecarver** [comment]
closing in favor of https://github.com/ibis-project/ibis/pull/4302

**Pull request closed without merging**